spark/spark-shims/src/main/scala/org/apache/zeppelin/spark/SparkShims.java

```diff
@@ -87,10 +101,31 @@ protected String getParagraphId(String jobgroupId) {
     return jobgroupId.substring(secondIndex + 1, jobgroupId.length());
   }
 
+  protected void setHttpClient(HttpClient httpClient) {
+    this.httpClient = httpClient;
+  }
+
   protected void buildSparkJobUrl(String sparkWebUrl, int jobId, Properties jobProperties) {
     String jobGroupId = jobProperties.getProperty("spark.jobGroup.id");
     String uiEnabled = jobProperties.getProperty("spark.ui.enabled");
     String jobUrl = sparkWebUrl + "/jobs/job?id=" + jobId;
+    // See ZEPPELIN-2221 which fixes an issue passing wrong parameters. It's related to YARN-6615.
+    HttpGet httpGetRequest = new HttpGet(jobUrl);
```

Comment:
Could you use org.apache.hadoop.util.VersionInfo to get the hadoop version info in SparkShims ? Otherwise invoking http request per spark job is very costly.

Reply:
I think so. But is `VersionInfo` getting the version dynamically?

Reply:
OH!!! I've tested it would get the current version. I'll fix it

Reply:
Fixed